### PR TITLE
KAFKA-17808 : Append connector name to admin client id - dlq sink reporter

### PR DESCRIPTION
**Pull request by muralibasani**
Jira : https://issues.apache.org/jira/browse/KAFKA-17808

### Committer Checklist (excluded from commit message)
- [X] Verify design and implementation 
- [X] Verify test coverage and CI build status
- [X] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java [hunk 1]
```diff
@@ -1005,7 +1005,7 @@ private List<ErrorReporter<ConsumerRecord<byte[], byte[]>>> sinkTaskReporters(Co
         if (topic != null && !topic.isEmpty()) {
             Map<String, Object> producerProps = baseProducerConfigs(id.connector(), "connector-dlq-producer-" + id, config, connConfig, connectorClass,
                                                                 connectorClientConfigOverridePolicy, kafkaClusterId);
-            Map<String, Object> adminProps = adminConfigs(id.connector(), "connector-dlq-adminclient-", config, connConfig, connectorClass, connectorClientConfigOverridePolicy, kafkaClusterId, ConnectorType.SINK);
+            Map<String, Object> adminProps = adminConfigs(id.connector(), "connector-dlq-adminclient-" + connConfig.get("connector.name"), config, connConfig, connectorClass, connectorClientConfigOverridePolicy, kafkaClusterId, ConnectorType.SINK);
```
Comment (muralibasani) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java:1008]: I see there is a method getConnectorNames() too. Not sure how we get the relevant connector name for this worker.

**Comment by chia7712** [comment]
@muralibasani Sorry for late review. Do you have free time to reopen PR and add tests?

**Comment by Yunyung** [comment]
@muralibasani Thanks for the patch. Do you have time to revisit this? Otherwise, I'd like to continue this PR and add some tests.

**Comment by muralibasani** [comment]
> @muralibasani Thanks for the patch. Do you have time to revisit this? Otherwise, I'd like to continue this PR and add some tests.

@Yunyung thank you and pls take over. It would be great.

**Pull request closed without merging**